ext/libclementine-tagreader/tagreader.cpp

```diff
@@ -872,108 +843,68 @@ bool TagReader::SaveFile(const QString& filename,
+  // if more tags are required propertymap() shall be used
 
-  if (TagLib::MPEG::File* file =
-          dynamic_cast<TagLib::MPEG::File*>(fileref->file())) {
-    TagLib::ID3v2::Tag* tag = file->ID3v2Tag(true);
-    SetTextFrame("TPOS",
-                 song.disc() <= 0 ? QString() : QString::number(song.disc()),
-                 tag);
-    SetTextFrame("TBPM",
-                 song.bpm() <= 0 - 1 ? QString() : QString::number(song.bpm()),
-                 tag);
-    SetTextFrame("TCOM", song.composer(), tag);
-    SetTextFrame("TIT1", song.grouping(), tag);
-    SetTextFrame("TOPE", song.performer(), tag);
-    SetUnsyncLyricsFrame(song.lyrics(), tag);
-    // Skip TPE1 (which is the artist) here because we already set it
-    SetTextFrame("TPE2", song.albumartist(), tag);
-    SetTextFrame("TCMP", song.compilation() ? QString::number(1) : QString(),
-                 tag);
-  } else if (TagLib::FLAC::File* file =
-                 dynamic_cast<TagLib::FLAC::File*>(fileref->file())) {
-    TagLib::Ogg::XiphComment* tag = file->xiphComment();
-    SetVorbisComments(tag, song);
-  } else if (TagLib::MP4::File* file =
-                 dynamic_cast<TagLib::MP4::File*>(fileref->file())) {
-    TagLib::MP4::Tag* tag = file->tag();
-    tag->itemListMap()["disk"] =
-        TagLib::MP4::Item(song.disc() <= 0 - 1 ? 0 : song.disc(), 0);
-    tag->itemListMap()["tmpo"] = TagLib::StringList(
-        song.bpm() <= 0 - 1 ? "0" : TagLib::String::number(song.bpm()));
-    tag->itemListMap()["\251wrt"] = TagLib::StringList(song.composer().c_str());
-    tag->itemListMap()["\251grp"] = TagLib::StringList(song.grouping().c_str());
-    tag->itemListMap()["\251lyr"] = TagLib::StringList(song.lyrics().c_str());
-    tag->itemListMap()["aART"] = TagLib::StringList(song.albumartist().c_str());
-    tag->itemListMap()["cpil"] =
-        TagLib::StringList(song.compilation() ? "1" : "0");
-  } else if (TagLib::APE::File* file =
-                 dynamic_cast<TagLib::APE::File*>(fileref->file())) {
-    saveApeTag(file->APETag(true));
-  } else if (TagLib::MPC::File* file =
-                 dynamic_cast<TagLib::MPC::File*>(fileref->file())) {
-    saveApeTag(file->APETag(true));
-  } else if (TagLib::WavPack::File* file =
-                 dynamic_cast<TagLib::WavPack::File*>(fileref->file())) {
-    saveApeTag(file->APETag(true));
-  }
+  bool ret = fileref->save();
 
-  // Handle all the files which have VorbisComments (Ogg, OPUS, ...) in the same
-  // way;
-  // apart, so we keep specific behavior for some formats by adding another
-  // "else if" block above.
-  if (TagLib::Ogg::XiphComment* tag =
-          dynamic_cast<TagLib::Ogg::XiphComment*>(fileref->file()->tag())) {
-    SetVorbisComments(tag, song);
+#ifdef Q_OS_LINUX
+  if (ret) {
+    // Linux: inotify doesn't seem to notice the change to the file unless we
+    // change the timestamps as well. (this is what touch does)
+    utimensat(0, QFile::encodeName(filename).constData(), nullptr, 0);
```

Comment:
Not sure why that would be. The Qt code that handles the inotify specifies the IN_MODIFY flag. And the modification of the file would be updating the timestamp as well. Not sure what games taglib would be playing to circumvent this.

Reply:
It's just a copy from former code. I would like to remove that.